lib/src/repo.rs

```diff
@@ -1664,6 +1664,7 @@ impl MutableRepo {
     }
 
     fn merge_view(&mut self, base: &View, other: &View) -> BackendResult<()> {
+        // TODO(ilyagr): It might be possible to use a merge join here
```

Comment:
I can remove this TODO. Perhaps this code will be clearer to me tomorrow.

Reply:
Yes, I think this can be merge join, and maybe we can use trivial merge function.

Then, it might make sense to add `refs::diff_named_commit_ids(..) -> impl Iterator<Item = (K, (Option<&CommitId>, Option<&CommitId>))>`. There will be two callers.

Reply:
I think I'll just reword the TODO for now, this cleanup can go into a separate PR (unless I suddenly feel like doing it immediately).

I do like it, if the working copies are returned from views a `<K, &CommitId>`, avoiding the conversion to RefTarget until the very end will simplify things.

Reply:
OK, I did actually do it, it does turn out quite nicely.

I'm only 90% sure the simplification in `repo.rs` is correct, PTAL. If it needs work, I'll split it out into a separate PR.